tests/modules/saml/src/Controller/ServiceProviderTest.php

```diff
@@ -185,6 +208,246 @@ public function testWithAuthIdWithoutEntity(): void
     }
 
 
+    /**
+     * @return array
+     */
+    public static function loginNotAuthenticatedDataProvider(): array
+    {
+        return [
+            'no query params' => [
+                [], // If no query params are provided, it should default to the RelayState
+                [
+                    'ReturnTo' => self::RELAY_STATE
+                ]
+            ],
+            'entityID=example.edu' => [
+                [
+                    'entityID' => 'example.edu',
+                ],
+                [
+                    'ReturnTo' =>  self::RELAY_STATE,
+                    'saml:idp' => 'example.edu'
+                ]
+            ],
+            'entityID=example.edu&forceAuthn=false&isPassive=false' => [
+                [
+                    'entityID' => 'example.edu',
+                    'forceAuthn' => 'false',
+                    'isPassive' => 'false'
+                ],
+                [
+                    'ReturnTo' => self::RELAY_STATE,
+                    'saml:idp' => 'example.edu',
+                    'ForceAuthn' => false,
+                    'isPassive' => false
+                ]
+            ],
+            'entityID=other.edu' => [
+                [
+                    'entityID' => 'other.edu',
+                ],
+                [
+                    'ReturnTo' =>  self::RELAY_STATE,
+                    'saml:idp' => 'other.edu'
+                ]
+            ],
+            'forceAuthn=true&target=/some/url' => [
+                [
+                    'target' => '/some/url',
+                    'forceAuthn' => 'true'
+                ],
+                [
+                    'ReturnTo' => 'http://localhost/some/url',
+                    'ForceAuthn' => true
+                ],
+                false
+            ],
+            'target=https://evil.com' => [
+                [
+                    'target' => 'https://evil.com'
+                ],
+                [],
+                true
+            ]
+        ];
+    }
+
+    /**
+     * Test Login without Session/ Not Authenticated
+     *
+     * @dataProvider loginNotAuthenticatedDataProvider
```

Comment:
Please use PHPUnit [attributes](https://docs.phpunit.de/en/10.5/writing-tests-for-phpunit.html#data-providers)-syntax instead

Reply:
Done